### PR TITLE
Fix return type of `polynomial.__eq__` to `numpy.bool_`

### PR DESCRIPTION
**Pull request by asi1024**
#4554

### CODE REVIEW

File: cupy/lib/_polynomial.pyx [hunk 1]
```diff
@@ -193,7 +193,7 @@ cdef class poly1d:
             raise NotImplementedError
         if self.coeffs.shape != other.coeffs.shape:
             return False
-        return bool((self.coeffs == other.coeffs).all())
+        return numpy.bool_((self.coeffs == other.coeffs).all().item())
```
Comment (toslunar) [inline cupy/lib/_polynomial.pyx:196]: I suppose we don't have to convert it via Python bool.
```suggestion
        return (self.coeffs == other.coeffs).all().get()[()]
```

**Review by toslunar — COMMENTED** [review]
Should `polynomial.__eq__` return a CPU value?  `cupy.allclose` returns `cupy.ndarray` even though `numpy.allclose` returns Python `bool`.

**Comment by asi1024** [comment]
I have two reasons for returning a CPU value as a result.
1. When the size of coeffs are different, it can return `False` as a CPU value immediately. I don't want to mix CPU and GPU values in return value depending on the sizes of inputs.
2. I think we don't need to pay so much attention to device-to-host synchronization in `cupy.poly1d`, because `cupy.poly1d` already requires many device-to-host synchronization mainly due to `trim_zeros` for coeffs.

**Review by toslunar — APPROVED** [review]
Then, LGTM

**Comment by toslunar** [comment]
pfnCI, test this please.

**Pull request merged**